packages/utils/node-resolver-rs/src/cache.rs

```diff
@@ -22,11 +22,14 @@ pub struct Cache {
   /// way to associate a lifetime with owned data stored in the same struct. We only vend temporary references
   /// from our public methods so this is ok for now. FrozenMap is an append only map, which doesn't require &mut
   /// to insert into. Since each value is in a Box, it won't move and therefore references are stable.
-  packages: RwLock<HashMap<PathBuf, Arc<Result<Arc<PackageJson>, ResolverError>>>>,
-  tsconfigs: RwLock<HashMap<PathBuf, Arc<Result<Arc<TsConfigWrapper>, ResolverError>>>>,
-  is_file_cache: DashMap<PathBuf, bool>,
-  is_dir_cache: DashMap<PathBuf, bool>,
-  realpath_cache: DashMap<PathBuf, Option<PathBuf>>,
+  packages: DashMap<PathBuf, Arc<Result<Arc<PackageJson>, ResolverError>>, gxhash::GxBuildHasher>,
+  tsconfigs:
+    DashMap<PathBuf, Arc<Result<Arc<TsConfigWrapper>, ResolverError>>, gxhash::GxBuildHasher>,
+  // In particular just the is_dir_cache spends around 8% of the time on a large project resolution
+  // hashing paths. Instead of using a hashmap we should try a trie here.
+  is_dir_cache: DashMap<PathBuf, bool, gxhash::GxBuildHasher>,
+  is_file_cache: DashMap<PathBuf, bool, gxhash::GxBuildHasher>,
```

Comment:
Something I found before but never got around to committing was that making the keys `OsString` instead of `PathBuf` made the hashing significantly faster. I think that should be equivalent as long as the path separators are normalized?

Reply:
Most of the hashing time is spent in the function body of the `Path` hash function (rather than the hasher).

So I think it might be faster to just use strings, but the hashes will be different if any path key isn't normalized yes.